### PR TITLE
[Admin] [Shipping] Sorting shipping methods

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Related tickets | -
| License         | MIT

### CODE REVIEW

File: features/shipping/managing_shipping_methods/adding_shipping_method.feature [hunk 1]
```diff
@@ -13,6 +13,7 @@ Feature: Adding a new shipping method
     Scenario: Adding a new shipping method with flat rate per shipment
         Given I want to create a new shipping method
         When I specify its code as "FED_EX_CARRIER"
+        And I specify its position as 0
```
Comment (pjedrzejewski) [inline features/shipping/managing_shipping_methods/adding_shipping_method.feature]: This should not be required.

File: features/shipping/managing_shipping_methods/sorting_shipping_methods_by_name_or_code.feature [hunk 2]
```diff
@@ -16,15 +16,17 @@ Feature: Sorting listed shipping methods
         And I am logged in as an administrator
 
     @ui
-    Scenario: Shipping methods are sorted by code in ascending order by default
-        When I am browsing shipping methods
+    Scenario: Shipping methods can be sorted by code in ascending order
+        Given I am browsing shipping methods
+        When I start sorting shipping methods by code
```
Comment (Zales0123) [inline features/shipping/managing_shipping_methods/sorting_shipping_methods_by_name_or_code.feature]: ``I start sorting`` assumes its a long process... I think ``I sort`` would be more appropriate.

File: src/Sylius/Bundle/ShippingBundle/Form/Type/ShippingMethodType.php [hunk 3]
```diff
@@ -73,6 +74,10 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'type' => 'sylius_shipping_method_translation',
                 'label' => 'sylius.form.shipping_method.translations',
             ])
+            ->add('position', IntegerType::class, [
```
Comment (Zales0123) [inline src/Sylius/Bundle/ShippingBundle/Form/Type/ShippingMethodType.php]: Why not ``'integer'``?

File: src/Sylius/Bundle/ShippingBundle/Resources/translations/messages.en.yml [hunk 4]
```diff
@@ -61,4 +61,5 @@ sylius:
             name: Name
             tax_category: Tax category
             translations: Translations
+            position: Position
```
Comment (Zales0123) [inline src/Sylius/Bundle/ShippingBundle/Resources/translations/messages.en.yml]: Alphabetical order R.I.P [*] 😢

File: features/shipping/managing_shipping_methods/adding_shipping_method.feature (hunk 1 above)
Comment (pamil) [inline features/shipping/managing_shipping_methods/adding_shipping_method.feature]: It checks that position can be set. Position 0 is the first element, empty position equals -1 which equals the last element.

File: src/Sylius/Bundle/ShippingBundle/Form/Type/ShippingMethodType.php (hunk 3 above)
Comment (pamil) [inline src/Sylius/Bundle/ShippingBundle/Form/Type/ShippingMethodType.php]: It's the recommended way for Symfony 2.8+, we shouldn't use deprecated features if we can (we can for `integer` but not for `sylius_shipping_category_choice`).

File: src/Sylius/Bundle/ShippingBundle/Resources/translations/messages.en.yml (hunk 4 above)
Comment (pamil) [inline src/Sylius/Bundle/ShippingBundle/Resources/translations/messages.en.yml]: Not a big deal anyway, we would need to sort the whole file.

File: features/shipping/managing_shipping_methods/sorting_shipping_methods_by_name_or_code.feature (hunk 2 above)
Comment (pamil) [inline features/shipping/managing_shipping_methods/sorting_shipping_methods_by_name_or_code.feature]: It's used both here and in products, I'd rather rename it in another PR as it is not a subject of this one.

**Comment by pjedrzejewski** [comment]
Thanks Kamil! :)

**Pull request merged**